qiskit_experiments/characterization/t1_experiment.py

```diff
@@ -0,0 +1,231 @@
+
+        return analysis_result, None
+
+    @staticmethod
+    def _fit_quality(fit_out, fit_err, chisq):
+        # pylint: disable = too-many-boolean-expressions
+        if (
+            abs(fit_out[0] - 1.0) < 0.1
+            and abs(fit_out[2]) < 0.1
+            and chisq < 3
+            and (fit_err[0] is None or fit_err[0] < 0.1)
+            and (fit_err[1] is None or fit_err[1] < fit_out[1])
+            and (fit_err[2] is None or fit_err[2] < 0.1)
+        ):
+            return "computer_good"
+        else:
+            return "computer_bad"
+
+
+class T1Experiment(BaseExperiment):
+    """T1 experiment class"""
+
+    __analysis_class__ = T1Analysis
+
+    def __init__(
+        self,
+        qubit: int,
+        delays: Union[List[float], np.array],
+        unit: Optional[str] = "us",
+        experiment_type: Optional[str] = None,
+    ):
+        """
+        Initialize the T1 experiment class
+
+        Args:
+            qubit: the qubit whose T1 is to be estimated
+            delays: delay times of the experiments
+            unit:Optional, unit of the duration. Supported units: 's', 'ms', 'us', 'ns', 'ps', 'dt'.
+            experiment_type: Optional, the experiment type string.
+
+        Raises:
+            ValueError: if the number of delays is smaller than 3
+        """
+        if len(delays) < 3:
+            raise ValueError("T1 experiment: number of delays must be at least 3")
+
+        self._delays = delays
+        self._unit = unit
+        super().__init__([qubit], experiment_type)
+
+    # pylint: disable=arguments-differ
+    def circuits(self, backend: Optional["Backend"] = None) -> List[QuantumCircuit]:
+        """
+        Return a list of experiment circuits
+
+        Args:
+            backend: Optional, a backend object
+
+        Returns:
+            The experiment circuits
+
+        Raises:
+            AttributeError: if unit is dt but dt parameter is missing in the backend configuration
+        """
+
+        if self._unit == "dt":
+            try:
+                dt_factor_in_sec = getattr(backend.configuration(), "dt")
+            except AttributeError as no_dt:
+                raise AttributeError("Dt parameter is missing in backend configuration") from no_dt
+
+        circuits = []
+
+        for delay in self._delays:
+            circ = QuantumCircuit(1, 1)
+            circ.x(0)
+            circ.barrier(0)
+            circ.delay(delay, 0, self._unit)
+            circ.barrier(0)
+            circ.measure(0, 0)
+
+            circ.metadata = {
+                "experiment_type": self._type,
+                "qubit": self.physical_qubits[0],
+                "delay": delay,
+                "unit": self._unit,
+            }
+
+            if self._unit == "dt":
+                circ.metadata["dt_factor_in_sec"] = dt_factor_in_sec
+
+            circuits.append(circ)
+
+        return circuits
+
+    @staticmethod
+    def generate_delays(t1_estimate, num_of_points):
```

Comment:
This is a bit of an odd static method for this class. If the intention is to use this to get input delays for the experiment, why not make it a static method that returns the appropriate T1Experiment object instead:

```python
@staticmethod
def from_t1_estimate(qubit, t1_estimate, num_of_points, unit="us"):
    delays = ...
    return T1Experiment(qubit, delays, unit=unit)
```

That said I am slightly opposed to adding a custom initializer like this to our first experiment since it sets a bad precedent for having experiments share a simple and common interface. I think it might be better to just to add the extra kwargs in the regular init for the experiment if this is something that will be used regularly, or just remove it if its not something that will be used regularly.

Reply:
@wshanks @oliverdial @coruscating Do you expect it to be used regularly?

Reply:
You have to be pretty careful setting your delay range based on previous T1 estimates so that outliers don't drive you nuts, but I guess this doesn't really decide how you estimated it.  The hard-coded 3 is probably not usually what you want, especially with parametrized units.  The value over the user just calling np.geomspace themselves seems minimal?

Reply:
@oliverdial What do you mean by "this doesn't really decide how you estimated it"? What's parametrized units? And the last sentence, I guess that it's not a question, but a statement that we'd better drop this function?

Reply:
Oops -- sorry, I wrote this too quickly I apologize.  There are a lot of ways to get in trouble picking a t1_estimate for this kind of thing, but this doesn't really bear on that so I don't think you have to worry about it.  Parameterized units -- if someone picked any unit other than "us" above (or didn't have a superconducting qubit), the 3 would become pretty inappropriate.  But if you let the person pass in min, max, and number of steps, and parameterize the 3 and 7, the function becomes almost a renamed copy of geomspace?

Reply:
On the question of it being used regularly, that is a little hard to say because I can't picture the workflow yet for running experiments interactively with qiskit-experiments. The way the old system works is that it tries to pull a t1guess-like variable from a global configuration (really the variable is "maximum delay for t1 experiments") and defaults to an arbitrary value if it can't find it. Then it generates the delays from that (it also looks for a "number of delays for t1 experiments" variable).

So this class could work similarly by making `delays` and `t1guess` optional inputs and calculating a default set of delays if they are not given. That sounds nice to me (not needing to create the delays in a separate step), but maybe with the way qiskit-experiments is used it will be easier to call another function to generate the delays first than it was in the old system, and then the extra complexity of creating default delays is not worth it.

Reply:
Summarizing the previous comments, we need to figure out:
1. Is there a way to set delays based only on an estimate and a number of delays? If min and max are also required, then, as @oliverdial comments, the benefit from using delays generation over a manual setting of delays is very small.
2. If yes, do we want it to be in the constructor or in a separate function? If in the constructor, how will the interface look like? In particular, what will be the relation between a T1 estimate given in the constructor (for delays generation) and a T1 estimate given as a `kwarg` parameter to the `run` function (to be passed to the analysis) ?